internal/pkg/template/repository/manager/manager_test.go

```diff
@@ -82,6 +90,90 @@ func TestRepository(t *testing.T) {
 	defer unlockFn2()
 }
 
+func TestRepositoryUpdate(t *testing.T) {
+	t.Parallel()
+	if runtime.GOOS == "windows" {
+		t.Skip("unstable on windows - random timeouts")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Copy the git repository to temp
+	tmpDir := t.TempDir()
+	assert.NoError(t, aferofs.CopyFs2Fs(nil, filepath.Join("test", "repository"), nil, tmpDir))
+	assert.NoError(t, os.Rename(filepath.Join(tmpDir, ".gittest"), filepath.Join(tmpDir, ".git")))
+	repo := model.TemplateRepository{
+		Type: model.RepositoryTypeGit,
+		Name: repository.DefaultTemplateRepositoryName,
+		URL:  fmt.Sprintf("file://%s", tmpDir),
+		Ref:  "main",
+	}
+
+	// Create manager
+	d := dependencies.NewMockedDeps(t)
+	m, err := manager.New(ctx, d, []model.TemplateRepository{repo})
+	assert.NoError(t, err)
+
+	// 1. update - no change
+	assert.NoError(t, <-m.Update(ctx))
+
+	// Modify git repository
+	runGitCommand(t, tmpDir, "reset", "--hard", "HEAD~1")
+
+	// 2. update - change
+	assert.NoError(t, <-m.Update(ctx))
+
+	// Check metrics
+	histBounds := []float64{0, 5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000} // ms
+	d.TestTelemetry().AssertMetrics(t,
+		[]metricdata.Metrics{
+			{
+				Name:        "keboola.go.templates.repository.update.duration",
+				Description: "Templates repository update duration.",
+				Unit:        "ms",
+				Data: metricdata.Histogram[float64]{
+					Temporality: 1,
+					DataPoints: []metricdata.HistogramDataPoint[float64]{
+						{
+							Count:  1,
+							Bounds: histBounds,
+							Attributes: attribute.NewSet(
+								attribute.String("repo_name", "keboola"),
+								attribute.String("repo_url", "file://<tmp_dir>"),
+								attribute.String("repo_ref", "main"),
+								attribute.Bool("is_success", true),
+								attribute.Bool("is_changed", false),
+							),
+						},
+						{
+							Count:  1,
+							Bounds: histBounds,
+							Attributes: attribute.NewSet(
+								attribute.String("repo_name", "keboola"),
+								attribute.String("repo_url", "file://<tmp_dir>"),
+								attribute.String("repo_ref", "main"),
+								attribute.Bool("is_success", true),
+								attribute.Bool("is_changed", true),
+							),
+						},
+					},
```

Comment:
Unit test, 1x ziadna zmena, 2x zmena.

Testuje sa to oproti Git tmpl directory `file://....`.